pkgs/development/tools/godot/default.nix

```diff
@@ -1,103 +1,146 @@
-{ stdenv
-, lib
+{ alsa-lib
 , fetchFromGitHub
-, scons
-, pkg-config
-, udev
+, freetype
+, installShellFiles
+, lib
+, libGLU
+, libpulseaudio
 , libX11
 , libXcursor
+, libXext
+, libXfixes
+, libXi
 , libXinerama
 , libXrandr
 , libXrender
-, libpulseaudio
-, libXi
-, libXext
-, libXfixes
-, freetype
 , openssl
-, alsa-lib
-, libGLU
-, zlib
+, pkg-config
+, scons
+, stdenv
+, udev
 , yasm
-, withUdev ? true
+, zlib
 }:
 
-let
-  options = {
-    touch = libXi != null;
-    pulseaudio = false;
-    udev = withUdev;
-  };
-in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (self: {
   pname = "godot";
   version = "3.4.4";
 
   src = fetchFromGitHub {
     owner = "godotengine";
     repo = "godot";
-    rev = "${version}-stable";
-    sha256 = "sha256-3AESLzqozi7Fc80u8Ml3ergZMkIhHy4tNlRe/3FsE6k=";
+    rev = self.passthru.githubRev;
+    sha256 = self.passthru.githubSha256;
+  };
+
+  passthru = {
+    githubRev = "${self.version}-stable";
+    githubSha256 = "sha256-3AESLzqozi7Fc80u8Ml3ergZMkIhHy4tNlRe/3FsE6k=";
+
+    sconsFlagSet = {
+      target = "release_debug";
+      platform = "x11";
+      tools = true;
+      touch = libXi != null;
+      pulseaudio = false;
+      udev = true;
+    };
+
+    buildDescription = null;
   };
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [ installShellFiles pkg-config scons ];
+
   buildInputs = [
-    scons
-    udev
+    alsa-lib
+    freetype
+    libGLU
+    libpulseaudio
     libX11
     libXcursor
+    libXext
+    libXfixes
+    libXi
     libXinerama
     libXrandr
     libXrender
-    libXi
-    libXext
-    libXfixes
-    freetype
     openssl
-    alsa-lib
-    libpulseaudio
-    libGLU
-    zlib
+    udev
     yasm
+    zlib
   ];
 
   patches = [ ./pkg_config_additions.patch ./dont_clobber_environment.patch ];
 
   enableParallelBuilding = true;
 
-  sconsFlags = "target=release_debug platform=x11";
-  preConfigure = ''
-    sconsFlags+=" ${
-      lib.concatStringsSep " "
-      (lib.mapAttrsToList (k: v: "${k}=${builtins.toJSON v}") options)
-    }"
-  '';
+  sconsFlags =
+    let
+      toSconsString = x: if builtins.isBool x then lib.boolToString x else toString x;
+    in
+      lib.mapAttrsToList (k: v: "${k}=${toSconsString v}") self.passthru.sconsFlagSet;
 
-  outputs = [ "out" "dev" "man" ];
+  shouldInstallBin = true;
+  shouldInstallShortcut = true;
+  shouldInstallHeaders = true;
+  shouldInstallManual = true;
+
+  outputs = []
+    ++ lib.optional (self.shouldInstallBin || self.shouldInstallShortcut) "out"
+    ++ lib.optional self.shouldInstallHeaders "dev"
+    ++ lib.optional self.shouldInstallManual "man";
+
+  # Where to install the godot bin files within the out directory.
```

Comment:
We don't need to repeat the stdenv documentation

Reply:
That's a variable I defined, not a stdenv one.  It's a variable containing a subpath within the `$out` directory where the binaries should be installed.  in godot/default.nix, it's simply `"bin"`, so the binaries are installed in `$out/bin`.  But in the two export-templates.nix files, it's installed in a different location per the godot build instructions, particularly `$out/share/godot/templates/${self.version}-stable`.

Reply:
I think we should rename this. Right now you can easily confuse it with a variable from stdenv. I don't have a good suggestion right now.

Reply:
I could rename it to godotInstallPath I guess, so it's clearly specific to these derivations and not some nixpkgs-general thing.